src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/config/ResourceServerConfiguration.kt

```diff
@@ -1,17 +1,38 @@
 package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.config
 
+import jakarta.servlet.Filter
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletRequest
+import jakarta.servlet.ServletResponse
+import jakarta.servlet.http.HttpServletRequest
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.security.authentication.AbstractAuthenticationToken
 import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
 import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.security.web.SecurityFilterChain
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter
 
 @Configuration
 @EnableWebSecurity
 @EnableMethodSecurity(prePostEnabled = true, proxyTargetClass = true)
 class ResourceServerConfiguration {
+
+  private class ActiveCaseLoadFilter : Filter {
+    override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain?) {
+      val httpRequest = request as? HttpServletRequest
+
+      val activeCaseload = httpRequest?.getHeader("Active-Caseload")
```

Comment:
It could be null? Do we want to handle that scenario?

Reply:
updated inline with suggestion, save accessing security context if its null